### PR TITLE
Gradle update

### PR DESCRIPTION
**Pull request by petardz**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by sbcd90 — APPROVED** [review]

**Review by eirsep — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -329,17 +329,17 @@ afterEvaluate {
         fileMode 0644
         dirMode 0755
 
-        requires('opensearch', versions.opensearch, EQUAL)
+        requires('elasticsearch-oss', versions.elasticsearch, EQUAL)
```
Comment (eirsep) [inline build.gradle:332]: why aer we changing this?

File: build.gradle [hunk 2]
```diff
@@ -329,17 +329,17 @@ afterEvaluate {
...
         packager = 'Amazon'
         vendor = 'Amazon'
         os = 'LINUX'
         prefix '/usr'
 
         license 'ASL-2.0'
-        maintainer 'OpenSearch Team <opensearch@amazon.com>'
+        maintainer 'OpenDistro for Elasticsearch Team <opendistro@amazon.com>'
```
Comment (eirsep) [inline build.gradle:339]: why are we changing this?

Comment (petardz) [reply build.gradle:339]: this was copied over from ISM. Let me revert i

Comment (petardz) [reply build.gradle]: reverted

File: build.gradle (hunk 1 above)
Comment (petardz) [reply build.gradle]: reverted

**Review by AWSHurneyt — APPROVED** [review]

File: build.gradle [hunk 3]
```diff
@@ -317,7 +317,7 @@ afterEvaluate {
     ospackage {
         packageName = "${name}"
         release = isSnapshot ? "0.1" : '1'
-        version = "${project.version}"
+        version = "${project.version}" - "-SNAPSHOT"
```
Comment (sbcd90) [inline build.gradle:320]: why do we need to change this?

**Review by sbcd90 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (petardz) [reply build.gradle:320]: reverted this change

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**